### PR TITLE
increase msrv and update dependencies 

### PR DESCRIPTION
**Pull request by Ruben2424**

### CODE REVIEW

File: h3-quinn/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,6 @@ quinn = { version = "0.10", default-features = false, features = [
     "futures-io",
 ] }
 quinn-proto = { version = "0.10", default-features = false }
-tokio-util = { version = "0.7.7" }
-futures = { version = "0.3.27" }
-tokio = { version = "1.28", features = ["io-util"], default-features = false }
+tokio-util = { version = "0.7.9" }
+futures = { version = "0.3.28" }
+tokio = { version = "1.33", features = ["io-util"], default-features = false }
```
Comment (seanmonstar) [inline h3-quinn/Cargo.toml:24]: Is such a newer version of Tokio required? If not, we can keep the lower number, so it works with older versions. Anyone who had a newer one still works too.

Comment (Ruben2424) [reply h3-quinn/Cargo.toml:24]: Not required i think. I will change that.

File: h3-quinn/Cargo.toml [hunk 2]
```diff
@@ -19,6 +19,6 @@ quinn = { version = "0.10", default-features = false, features = [
     "futures-io",
 ] }
 quinn-proto = { version = "0.10", default-features = false }
-tokio-util = { version = "0.7.7" }
-futures = { version = "0.3.27" }
-tokio = { version = "1.28", features = ["io-util"], default-features = false }
+tokio-util = { version = "0.7.9" }
+futures = { version = "0.3.28" }
+tokio = { version = "1", features = ["io-util"], default-features = false }
```
Comment (seanmonstar) [inline h3-quinn/Cargo.toml:24]: Is it true we work on 1.0? Usually we keep the minimum version that works. I suspect we use something that was newer than that. It's possible to check with `minimal-versions`.

Comment (Ruben2424) [reply h3-quinn/Cargo.toml:24]: Hmm `cargo minimal-versions check --workspace` is successful on my machine. Is there something else to check?

Comment (seanmonstar) [reply h3-quinn/Cargo.toml:24]: Alrighty, sounds good!

**Review by seanmonstar — APPROVED** [review]

**Pull request merged**